src/coreclr/jit/objectalloc.cpp

```diff
@@ -1799,6 +1817,15 @@ void ObjectAllocator::UpdateAncestorTypes(GenTree* tree, ArrayStack<GenTree*>* p
                 keepChecking = true;
                 break;
 
+            case GT_SUB:
+                if (parent->TypeGet() != newType)
+                {
+                    parent->ChangeType(newType);
+                    ++parentIndex;
+                    keepChecking = true;
+                }
```

Comment:
Is it correct to keep retyping ancestors if the existing `GT_SUB` was not a pointer-result (e.g. `byref - byref => TYP_I_IMPL`)? I get it for the `&byref - int` case).

Reply:
Yeah we should stop propagation.

Reply:
Actually this already happens, the only possibilities here are
```
parent    new
-----     ----
int       int
byref     byref
byref     int
```
so we only propagate in the byref->int case.

I will add some comments/asserts to make this clearer.